exotations/solvers/exotica_time_indexed_rrt_connect_solver/src/time_indexed_rrt_connect.cpp

```diff
@@ -1,55 +1,53 @@
+//
+//  * Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of  nor the names of its contributors may be used to
+//    endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
 
 #include <exotica_time_indexed_rrt_connect_solver/time_indexed_rrt_connect.h>
 #include <ompl/geometric/planners/rrt/RRTConnect.h>
 #include <ompl/util/RandomNumbers.h>
 
-REGISTER_MOTIONSOLVER_TYPE("TimeIndexedRRTConnect", exotica::TimeIndexedRRTConnect)
+REGISTER_MOTIONSOLVER_TYPE("TimeIndexedRRTConnectSolver", exotica::TimeIndexedRRTConnectSolver)
 
 namespace exotica
 {
-OMPLTimeIndexedRNStateSpace::OMPLTimeIndexedRNStateSpace(TimeIndexedSamplingProblem_ptr &prob, TimeIndexedRRTConnectInitializer init) : ompl::base::CompoundStateSpace(), prob_(prob)
+OMPLTimeIndexedRNStateSpace::OMPLTimeIndexedRNStateSpace(TimeIndexedSamplingProblem_ptr &prob, TimeIndexedRRTConnectSolverInitializer init) : ompl::base::CompoundStateSpace(), prob_(prob)
 {
     setName("OMPLTimeIndexedRNStateSpace");
     unsigned int dim = prob->N;
     addSubspace(ompl::base::StateSpacePtr(new ompl::base::RealVectorStateSpace(dim)), 1.0);
     ompl::base::RealVectorBounds bounds(dim);
     for (int i = 0; i < dim; i++)
     {
-        bounds.setHigh(i, prob->getBounds()[i + dim]);
-        bounds.setLow(i, prob->getBounds()[i]);
+        bounds.setHigh(i, prob->GetBounds()[i + dim]);
+        bounds.setLow(i, prob->GetBounds()[i]);
     }
     getSubspace(0)->as<ompl::base::RealVectorStateSpace>()->setBounds(bounds);
     addSubspace(ompl::base::StateSpacePtr(new ompl::base::TimeStateSpace), 1.0);
-    getSubspace(1)->as<ompl::base::TimeStateSpace>()->setBounds(0, prob_->T);
+    getSubspace(1)->as<ompl::base::TimeStateSpace>()->setBounds(0, prob_->GetGoalTime());
```

Comment:
Is this a bug fix? `T` is the number of timesteps (or not?!) while goal time is something else. I think this is a hidden fix.

Reply:
Unresolved

Reply:
This is something we should be having a look at. It is definitely different now than before - GetGoalTime != GetT